### PR TITLE
confusion_matrix

### PR DESCRIPTION
**Pull request by ShawnXuan**

### CODE REVIEW

File: finetune.py [hunk 1]
```diff
@@ -218,6 +219,12 @@ def save_model(subdir):
     val_loader = DataLoader(val_dataset, batch_size=args.batch_size)
     val_prefetched = [batch for batch in val_loader]
 
+    if flow.env.get_rank() == 0:
+        filepath = os.path.join(args.output, "val_imagepath_label.pkl")
+        with open(filepath, "wb") as f:
+            pickle.dump(val_dataset.samples, f, protocol=pickle.HIGHEST_PROTOCOL)
+            print(f"Saving validation image file path and label to {filepath}")
+
```
Comment (ShawnXuan) [inline finetune.py:227]: 这里保存了图片文件名和图片类别id

```bash
[('./classes_23/val/n09193705/ILSVRC2012_val_00000002.JPEG', 0),
...
('./classes_23/val/n12998815/ILSVRC2012_val_00049500.JPEG', 22)]
```

File: finetune.py [hunk 2]
```diff
@@ -235,7 +242,11 @@ def save_model(subdir):
...
         metrics = eval(model, val_prefetched)
-        r0_print(f"epoch {epoch}", metrics)
+        r0_print(f"epoch {epoch}", metrics["accuarcy"])
         if args.save_snapshot:
             subdir = f"snapshot_epoch{epoch}_acc{metrics['accuarcy']}"
             save_model(subdir)
+            metric_file = f"metric_epoch{epoch}_acc{metrics['accuarcy']}.pkl"
+            with open(metric_file, "wb") as f:
+                pickle.dump(metrics, f, protocol=pickle.HIGHEST_PROTOCOL)
+
```
Comment (ShawnXuan) [inline finetune.py:252]: 这里保存了指标等信息，是一个dict对象，包括：
{"accuarcy": metric, "confusion_matrix": cm, "labels": labels, "preds": preds}

File: infer.py [hunk 3]
```diff
@@ -77,10 +72,11 @@ def read_and_transform(filepath):
     state_dict = flow.load(args.snapshot)
     model.load_state_dict(state_dict, strict=True)
     model.to("cuda")
+    model.eval()
```
Comment (ShawnXuan) [inline infer.py:75]: 这句是必须的
